### PR TITLE
Fixed hashtag search for DB search

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Fixes an issue where certain hashtags are not searchable when using database search.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51196

#### Release Note
```release-note
Fixes an issue where certain hashtags are not searchable when using database search.
```

### CODE REVIEW

File: server/channels/store/searchtest/post_layer.go [hunk 1]
```diff
@@ -192,6 +192,11 @@ var searchPostStoreTests = []searchTest{
 		Fn:   testSearchHashtagWithUnderscores,
 		Tags: []string{EngineAll},
 	},
+	{
+		Name: "Should be able to search by hashtags with dashses and numbers",
```
Comment (kostaspt) [inline server/channels/store/searchtest/post_layer.go:196]: 🙃 
```suggestion
		Name: "Should be able to search by hashtags with dashes and numbers",
```

**Review by kostaspt — APPROVED** [review]

File: server/channels/store/searchtest/post_layer.go (hunk 1 above)
Comment (harshilsharma63) [reply server/channels/store/searchtest/post_layer.go:196]: Fixed :P

**Review by nevyangelova — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
Approved based on previews reviews to unlock dev review

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes @harshilsharma63 ,
e2e-tests have passed.

**Pull request merged**